src/java/org/apache/cassandra/streaming/StreamSession.java

```diff
@@ -127,6 +134,9 @@
 {
     private static final Logger logger = LoggerFactory.getLogger(StreamSession.class);
 
+    // for test purpose to record received message and state transition
+    public static volatile MessageStateSink sink = new MessageStateSink();
```

Comment:
Why is this `volatile` rather than `final`?

Reply:
right. `final` is sufficient